deluge/ui/web/server.py

```diff
@@ -431,8 +441,19 @@ class TopLevel(resource.Resource):
 
     def __init__(self):
         resource.Resource.__init__(self)
+        gettext_js_exists = os.path.exists(
+            os.path.join(
+                os.path.join(os.path.dirname(__file__), 'js'),
```

Comment:
`os.path.join` can take any number of args and join them so only one needed. Also `isfile` is preferred to `exists` as we are definitely looking for a file. 

Also there is the `rpath` function to simplify generating absolute script paths. So you can remove this code and just use:

```python
if os.path.isfile(rpath('js', 'gettext.js'):
   self.putChild ...
```